tests/test_arrow_dataset.py

```diff
@@ -1362,14 +1362,12 @@ def test_map_input_columns(self, in_memory):
         with tempfile.TemporaryDirectory() as tmp_dir:
             with self._create_dummy_dataset(in_memory, tmp_dir, multiple_columns=True) as dset:
                 with dset.map(lambda col_1: {"label": col_1 % 2}, input_columns="col_1") as mapped_dset:
-                    self.assertEqual(mapped_dset[0].keys(), {"col_1", "col_2", "col_3", "label"})
+                    self.assertEqual(mapped_dset[0].keys(), {"col_1", "label"})
```

Comment:
Why don't you remove completely the test `test_map_input_columns`, introduced in #4971?

Reply:
Unfortunately input_columns wasn't tested before this PR, so I think it's nice to keep this test, but with the correct behavior (i.e. don't keep all the columns)